### PR TITLE
chore: correct named .json import per TypeScript 5.7

### PR DESCRIPTION
**Pull request by pjonsson**
## PR Checklist

- [ ] Addresses an existing open issue: fixes #000
- [ ] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/TypeStat/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [X] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/TypeStat/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Import the entire config and
select version out of that instead
of importing the version.

### CODE REVIEW

**Comment by pjonsson** [comment]
I wanted to see if the CI would pass, but apparently that requires approval, so marking this as ready for review. The `pnpm tsc` passes locally in the node:20 container for me.

File: src/cli/runCli.test.ts [hunk 1]
```diff
@@ -1,9 +1,13 @@
 import { describe, expect, it, vi } from "vitest";
 
-import { version } from "../../package.json";
+// Used to be "import { version } ..", but bug in TypeScript 5.7:
+// https://github.com/microsoft/TypeScript/issues/60589#issuecomment-2501614830
+import packageConfig from "../../package.json" with { type: "json" };
```
Comment (JoshuaKGoldberg) [inline src/cli/runCli.test.ts:5]: I don't think a bug in TypeScript. The repo isn't on Node16 resolution:

https://github.com/JoshuaKGoldberg/TypeStat/blob/6bfc04441f72cafbe4625cd4be0c610dfdff493e/tsconfig.json#L6-L7

What errors are you seeing? I don't have any issues running `npx typestat` locally.

Comment (pjonsson) [reply src/cli/runCli.test.ts:5]: You can see the error in the CI logs for the type checking job if you click the red cross on the renovatebot commit in this PR:
```
Error: src/cli/runCli.test.ts(3,10): error TS1544: Named imports from a JSON file into an ECMAScript module are not allowed when 'module' is set to 'NodeNext'.
Error: src/cli/runCli.test.ts(3,25): error TS1543: Importing a JSON file into an ECMAScript module requires a 'type: "json"' import attribute when 'module' is set to 'NodeNext'.
```

Comment (JoshuaKGoldberg) [reply src/cli/runCli.test.ts:5]: 🤔 but type checking on `main` is passing, https://github.com/JoshuaKGoldberg/TypeStat/actions/runs/12097459370/job/33732857242

Comment (pjonsson) [reply src/cli/runCli.test.ts:5]: Yeah, because main uses TypeScript 5.4.3, but this PR is on top of #1725 which bumps TypeScript to 5.7.2.

Comment (JoshuaKGoldberg) [reply src/cli/runCli.test.ts:5]: Oh gotcha! Thanks, I'd missed that.

I need to context switch away from TypeStat for the next day or two, but will take a look when I can come back. 👍

Comment (pjonsson) [reply src/cli/runCli.test.ts:5]: Sure, I'm not in a hurry with this, I just figured eventually you would need to upgrade to TypeScript 5.7.

Comment (JoshuaKGoldberg) [reply src/cli/runCli.test.ts:5]: Ok coming back: it's intentional that we can't used named specifies with `.json` imports. This is TypeScript more correctly adhering to ECMAScript Modules. 🚀

File: src/cli/runCli.test.ts [hunk 2]
```diff
@@ -1,9 +1,13 @@
 import { describe, expect, it, vi } from "vitest";
 
-import { version } from "../../package.json";
+// Used to be "import { version } ..", but bug in TypeScript 5.7:
+// https://github.com/microsoft/TypeScript/issues/60589#issuecomment-2501614830
```
Comment (JoshuaKGoldberg) [inline src/cli/runCli.test.ts:4]: Since TypeScript is actually in the right here:

```suggestion
```

**Pull request merged**